sympy/stats/frv_types.py

```diff
@@ -646,3 +650,200 @@ def Rademacher(name):
 
     """
     return rv(name, RademacherDistribution)
+
+class IdealSolitonDistribution(SingleFiniteDistribution):
+    _argnames = ('k',)
+
+    @staticmethod
+    def check(k):
+         _value_check(k.is_integer and k.is_positive,
+                    "'k' must be a positive integer.")
+
+    @property
+    def low(self):
+        return S.One
+
+    @property
+    def high(self):
+        return self.k
```

Comment:
Do we need this? I don't remember if there were such properties in other distribution classes.

Reply:
Yes, discrete random variables have them defined, so I added them